scipy/interpolate/_interpolate.py

```diff
@@ -2562,6 +2589,63 @@ def _evaluate_nearest(self, indices, norm_distances, out_of_bounds):
                    for i, yi in zip(indices, norm_distances)]
         return self.values[tuple(idx_res)]
 
+    @staticmethod
+    def _validate_not_complex(input_value, method):
+        if np.iscomplexobj(input_value):
+            raise ValueError(f"method {method} does not support complex "
+                             f"values. Use 'linear' or 'nearest'.")
+
+    def _validate_grid_dimensions(self, points, method):
+        k = self.SPLINE_DEGREE_MAP[method]
+        for i, point in enumerate(points):
+            ndim = len(np.atleast_1d(point))
+            if ndim <= k:
+                raise ValueError(f"There are {ndim} points in dimension {i},"
+                                 f" but method {method} requires at least "
+                                 f" {k+1} points per dimension.")
+
+    def _evaluate_spline(self, values, xi, spline_degree):
+        # ensure xi is 2D list of points to evaluate
+        if xi.ndim == 1:
+            xi = xi.reshape((1, xi.size))
+        m, n = xi.shape
+
+        # Non-stationary procedure: difficult to vectorize this part entirely
+        # into numpy-level operations. Unfortunately this requires explicit
+        # looping over each point in xi.
+
+        # can at least vectorize the first pass across all points in the
+        # last variable of xi.
+        last_dim = n - 1
+        first_values = self._do_spline_fit(self.grid[last_dim],
+                                           values,
+                                           xi[:, last_dim],
+                                           spline_degree)
+
+        # the rest of the dimensions have to be on a per point-in-xi basis
+        result = np.empty(m)
```

Comment:
For complex-valued `values`, this will need a `dtype=self.values.dtype` or some such